packages/hardhat-viem/src/internal/errors.ts

```diff
@@ -44,3 +44,21 @@ You can find a list of supported networks here: https://viem.sh/docs/clients/cha
     );
   }
 }
+
+export class DefaultWalletClientNotFoundError extends HardhatViemError {
+  constructor() {
+    super(
+      `Default wallet client not found. You can set the wallet client by passing it as a parameter in the deployContract function:
```

Comment:
Maybe something like:

```suggestion
      `Default wallet client not found. This can happen if no accounts were configured for this network (network: '${networkName}').
      
Alternatively, you can set a custom wallet client by passing it as a parameter in the deployContract function:
```

Of course this means receiving the network name in the constructor.